sources/models/src/modeled_types/shared.rs

```diff
@@ -1130,3 +1133,44 @@ mod test_valid_kmod_key {
         assert!(KmodKey::try_from(vec!["z"; KMOD_KEY_LENGTH + 1].join("")).is_err());
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// Input value that needs to be a positive value, but should not be greater
+/// than an i32::MAX.
+#[derive(Debug, PartialEq, Scalar)]
+pub struct PositiveInteger {
+    inner: i32,
```

Comment:
You might be able to make this work for `T: PartionOrd + SomethingAboutTheParse`, but maybe we don't have any `i16`, `i64`, or whatever.

Reply:
I blame VS Code.

Reply:
I thought about making it more flexible, but the problem with that is then the upper bounds could be too high.

There are some settings that take `i64` at least. But if I understand right, and please correct me if I'm wrong, if we made it more flexible it would make it possible to set an `i64` value for a field that eventually has to go into an `i32`. So I was thinking it would be safer to just be more verbose and have `PositiveInteger64`, `PositiveInteger16`, etc., as needed for whatever fields we want to contrain.

Reply:
I would have to try it out to see if I could make it work with type `T`. No problem, it can be changed to generic later and the change would be an implementation detail.